plugins/modules/dcnm_links.py

```diff
@@ -0,0 +1,3326 @@
+            intra-fabric link. If this parameter is different from 'src_fabric', then the link is considered
+            inter-fabric link.
+        type: str
+        required: true
+      src_device:
+        description:
+          - IP address or DNS name of the source switch which is part of the link being configured.
+        type: list
+        required: true
+      dst_device:
+        description:
+          - IP address or DNS name of the destination switch which is part of the link being configured.
+        type: str
+        required: true
+      src_interface:
+        description:
+          - Interface on the source device which is part of the link being configured.
+        type: str
+        required: true
+      dst_interface:
+        description:
+          - Interface on the destination device which is part of the link being configured.
+        type: str
+        required: true
+      template:
+        description:
+          - Name of the template that is applied on the link being configured.
+          - The last 3 template choices are applicable for inter-fabric links and the others
+            are applicable for intra-fabric links.
+          - This parameter is required only for 'merged' and 'replaced' states. It is
+          - optional for other states.
+        type: str
+        required: true
+        choices:
+          - int_intra_fabric_ipv6_link_local(intra-fabric)
+          - int_intra_fabric_num_link (intra-fabric)
+          - int_intra_fabric_unnum_link (intra-fabric)
+          - int_intra_vpc_peer_keep_alive_link (intra-fabric)
+          - int_pre_provision_intra_fabric_link (intra-fabric)
+          - ios_xe_int_intra_fabric_num_link (intra-fabric)
+          - ext_fabric_setup (inter-fabric)
+          - ext_multisite_underlay_setup (inter-fabric)
+          - ext_evpn_multisite_overlay_setup (inter-fabric)
+      profile:
+        description:
+          - Additional link related parameters that must be included while creating links.
+        suboptions:
+          peer1_ipv4_address:
+            description:
+              - IPV4 address of the source interface.
+              - This parameter is optional if the underlying fabric is ipv6 enabled.
+              - This parameter is required only if template is 'int_intra_fabric_num_link' or
+                'ios_xe_int_intra_fabric_num_link' or 'int_intra_vpc_peer_keep_alive_link'.
+            type: str
+            required: true
+          peer2_ipv4_address:
+            description:
+              - IPV4 address of the destination interface.
+              - This parameter is optional if the underlying fabric is ipv6 enabled.
+              - This parameter is required only if template is 'int_intra_fabric_num_link' or
+                'ios_xe_int_intra_fabric_num_link' or 'int_intra_vpc_peer_keep_alive_link'.
+            type: str
+            required: true
+          peer1_ipv6_address:
```

Comment:
Even in GUI this is optional if the fabric is not IPV6 enabled.  Since this is optional param, user can still configure the IPv6 addresses if that is the only address available